### ISSUE TITLE
[package] boost/1.75.0: Link error duplicate symbols on MSVC

### ISSUE THREAD

**Issue by jprochazk** [conan-io/conan-center-index#4008]
### Package and Environment Details
  * Package Name/Version: **boost/1.75.0**
  * Operating System+version: **Windows 10**
  * Compiler+version: **Visual Studio 16.8.3**
  * Conan version: **conan 1.32.1**
  * Python version: **Python 3.7.2**

### Conan profile
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.version=16
build_type=Release
[options]
[build_requires]
[env]
```

### Steps to reproduce
Build any of the [asio coroutine examples](https://www.boost.org/doc/libs/1_75_0/doc/html/boost_asio/examples/cpp17_examples.html) with latest MSVC and Boost 1.75.0. It should compile, but linking will fail:
```
error LNK2005: "void __cdecl boost::asio::dummy_return<void>(void)" (??$dummy_return@X@asio@boost@@YAXXZ) already defined
```

The fix already exists (https://github.com/chriskohlhoff/asio/pull/584), but it hasn't been merged yet. I see that the Boost recipe has some patches already, would it be possible to apply a patch to fix this for version 1.75.0?https://gist.github.com/jprochazk/ecb261bb99489a3d0fef03a4ae27c064

**Comment by jprochazk** [comment]
fixed by https://github.com/chriskohlhoff/asio/pull/584#issuecomment-752268548, but it still needs a PR here

**Comment by obi0ne** [comment]
Is this fixed on v1.77 ?

**Comment by jprochazk** [comment]
> Is this fixed on v1.77 ?

Yes
